dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,519 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+
+import itertools
+
+from typing import List, Tuple, Dict
+
+import dace
+from dace import registry, symbolic, nodes, StorageType, Language
+from dace.config import Config
+from dace.codegen.targets.target import TargetCodeGenerator
+from dace.codegen.prettycode import CodeIOStream
+from dace.codegen.targets.framecode import DaCeCodeGenerator
+from dace.codegen.dispatcher import TargetDispatcher
+from dace.sdfg.graph import MultiConnectorEdge
+from dace.sdfg.state import StateSubgraphView
+from dace.sdfg import find_input_arraynode, find_output_arraynode
+from dace.codegen.codeobject import CodeObject
+
+
+@registry.autoregister_params(name='rtl')
+class RTLCodeGen(TargetCodeGenerator):
+    """ RTL Code Generator (SystemVerilog) """
+
+    title = 'RTL'
+    target_name = 'rtl'
+    language = dace.Language.RTL
+
+    def __init__(self,
+                 frame_codegen: DaCeCodeGenerator,
+                 sdfg: dace.SDFG):
+        # store reference to sdfg
+        self.sdfg: dace.SDFG = sdfg
+        # store reference to frame code generator
+        self.frame: DaCeCodeGenerator = frame_codegen
+        # get dispatcher to register callbacks for allocation/nodes/.. code generators
+        self.dispatcher: TargetDispatcher = frame_codegen.dispatcher
+        # register node dispatcher -> generate_node(), predicate: process tasklets only
+        self.dispatcher.register_node_dispatcher(self,
+                                                 lambda sdfg, node: isinstance(node,
+                                                                               nodes.Tasklet) and node.language == Language.RTL)
+        # register all cpu type copies
+        cpu_storage_types = [StorageType.CPU_Pinned, StorageType.CPU_Heap, StorageType.CPU_ThreadLocal,
+                             StorageType.Register, StorageType.Default]
```

Comment:
This seems fishy, the RTL codegen should never touch CPU memory. I guess this is because we only have verilator in isolation right now, but can't we at least register ANY storage type when the predicate of RTL tasklet is fulfilled?

Reply:
I just checked, the memlet to the tasklet edge for example are of type `StorageType.CPU_Heap` and `StorageType.Register`, which is why i do have to register. Sure, that is probably even cleaner to register all types of storages. I will do this.